payu/experiment.py

```diff
@@ -514,19 +518,18 @@ def run(self, *user_flags):
             # Update MPI library module (if not explicitly set)
             # TODO: Check for MPI library mismatch across multiple binaries
             if mpi_module is None:
-                mpi_module = envmod.lib_update(
-                    model.exec_path_local,
+                envmod.lib_update(
+                    model.config.get('required_libs'),
                     'libmpi.so'
                 )
 
             model_prog = []
 
-            if mpi_module.startswith('openmpi'):
-                # Our MPICH wrapper does not support a working directory flag
-                model_prog.append('-wdir {0}'.format(model.work_path))
-            elif self.config.get('scheduler') == 'slurm':
+            wdir_arg = '-wdir'
+            if self.config.get('scheduler') == 'slurm':
                 # Slurm's launcher controls the working directory
```

Comment:
I've never really understood this comment. Probably better to just do

```suggestion
                # Option to set the working directory differs in slurm
```